File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-ird-5-4/FxLinkedNotionalSchedule.cpp
// FxLinkedNotionalSchedule.cpp 
#include "FxLinkedNotionalSchedule.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

FxLinkedNotionalSchedule::FxLinkedNotionalSchedule(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //constantNotionalScheduleReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* constantNotionalScheduleReferenceNode = xmlNode->FirstChildElement("constantNotionalScheduleReference");

   if(constantNotionalScheduleReferenceNode){constantNotionalScheduleReferenceIsNull_ = false;}
   else{constantNotionalScheduleReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- constantNotionalScheduleReferenceNode , address : " << constantNotionalScheduleReferenceNode << std::endl;
   #endif
   if(constantNotionalScheduleReferenceNode)
   {
      if(constantNotionalScheduleReferenceNode->Attribute("href") || constantNotionalScheduleReferenceNode->Attribute("id"))
      {
          if(constantNotionalScheduleReferenceNode->Attribute("id"))
          {
             constantNotionalScheduleReferenceIDRef_ = constantNotionalScheduleReferenceNode->Attribute("id");
             constantNotionalScheduleReference_ = boost::shared_ptr<NotionalReference>(new NotionalReference(constantNotionalScheduleReferenceNode));
             constantNotionalScheduleReference_->setID(constantNotionalScheduleReferenceIDRef_);
             IDManager::instance().setID(constantNotionalScheduleReferenceIDRef_,constantNotionalScheduleReference_);
          }
          else if(constantNotionalScheduleReferenceNode->Attribute("href")) { constantNotionalScheduleReferenceIDRef_ = constantNotionalScheduleReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { constantNotionalScheduleReference_ = boost::shared_ptr<NotionalReference>(new NotionalReference(constantNotionalScheduleReferenceNode));}
   }

   //initialValueNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* initialValueNode = xmlNode->FirstChildElement("initialValue");

   if(initialValueNode){initialValueIsNull_ = false;}
   else{initialValueIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- initialValueNode , address : " << initialValueNode << std::endl;
   #endif
   if(initialValueNode)
   {
      if(initialValueNode->Attribute("href") || initialValueNode->Attribute("id"))
      {
          if(initialValueNode->Attribute("id"))
          {
             initialValueIDRef_ = initialValueNode->Attribute("id");
             initialValue_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(initialValueNode));
             initialValue_->setID(initialValueIDRef_);
             IDManager::instance().setID(initialValueIDRef_,initialValue_);
          }
          else if(initialValueNode->Attribute("href")) { initialValueIDRef_ = initialValueNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { initialValue_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(initialValueNode));}
   }

   //varyingNotionalCurrencyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* varyingNotionalCurrencyNode = xmlNode->FirstChildElement("varyingNotionalCurrency");

   if(varyingNotionalCurrencyNode){varyingNotionalCurrencyIsNull_ = false;}
   else{varyingNotionalCurrencyIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- varyingNotionalCurrencyNode , address : " << varyingNotionalCurrencyNode << std::endl;
   #endif
   if(varyingNotionalCurrencyNode)
   {
      if(varyingNotionalCurrencyNode->Attribute("href") || varyingNotionalCurrencyNode->Attribute("id"))
      {
          if(varyingNotionalCurrencyNode->Attribute("id"))
          {
             varyingNotionalCurrencyIDRef_ = varyingNotionalCurrencyNode->Attribute("id");
             varyingNotionalCurrency_ = boost::shared_ptr<Currency>(new Currency(varyingNotionalCurrencyNode));
             varyingNotionalCurrency_->setID(varyingNotionalCurrencyIDRef_);
             IDManager::instance().setID(varyingNotionalCurrencyIDRef_,varyingNotionalCurrency_);
          }
          else if(varyingNotionalCurrencyNode->Attribute("href")) { varyingNotionalCurrencyIDRef_ = varyingNotionalCurrencyNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { varyingNotionalCurrency_ = boost::shared_ptr<Currency>(new Currency(varyingNotionalCurrencyNode));}
   }

   //varyingNotionalFixingDatesNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* varyingNotionalFixingDatesNode = xmlNode->FirstChildElement("varyingNotionalFixingDates");

   if(varyingNotionalFixingDatesNode){varyingNotionalFixingDatesIsNull_ = false;}
   else{varyingNotionalFixingDatesIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- varyingNotionalFixingDatesNode , address : " << varyingNotionalFixingDatesNode << std::endl;
   #endif
   if(varyingNotionalFixingDatesNode)
   {
      if(varyingNotionalFixingDatesNode->Attribute("href") || varyingNotionalFixingDatesNode->Attribute("id"))
      {
          if(varyingNotionalFixingDatesNode->Attribute("id"))
          {
             varyingNotionalFixingDatesIDRef_ = varyingNotionalFixingDatesNode->Attribute("id");
             varyingNotionalFixingDates_ = boost::shared_ptr<RelativeDateOffset>(new RelativeDateOffset(varyingNotionalFixingDatesNode));
             varyingNotionalFixingDates_->setID(varyingNotionalFixingDatesIDRef_);
             IDManager::instance().setID(varyingNotionalFixingDatesIDRef_,varyingNotionalFixingDates_);
          }
          else if(varyingNotionalFixingDatesNode->Attribute("href")) { varyingNotionalFixingDatesIDRef_ = varyingNotionalFixingDatesNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { varyingNotionalFixingDates_ = boost::shared_ptr<RelativeDateOffset>(new RelativeDateOffset(varyingNotionalFixingDatesNode));}
   }

   //fxSpotRateSourceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* fxSpotRateSourceNode = xmlNode->FirstChildElement("fxSpotRateSource");

   if(fxSpotRateSourceNode){fxSpotRateSourceIsNull_ = false;}
   else{fxSpotRateSourceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- fxSpotRateSourceNode , address : " << fxSpotRateSourceNode << std::endl;
   #endif
   if(fxSpotRateSourceNode)
   {
      if(fxSpotRateSourceNode->Attribute("href") || fxSpotRateSourceNode->Attribute("id"))
      {
          if(fxSpotRateSourceNode->Attribute("id"))
          {
             fxSpotRateSourceIDRef_ = fxSpotRateSourceNode->Attribute("id");
             fxSpotRateSource_ = boost::shared_ptr<FxSpotRateSource>(new FxSpotRateSource(fxSpotRateSourceNode));
             fxSpotRateSource_->setID(fxSpotRateSourceIDRef_);
             IDManager::instance().setID(fxSpotRateSourceIDRef_,fxSpotRateSource_);
          }
          else if(fxSpotRateSourceNode->Attribute("href")) { fxSpotRateSourceIDRef_ = fxSpotRateSourceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { fxSpotRateSource_ = boost::shared_ptr<FxSpotRateSource>(new FxSpotRateSource(fxSpotRateSourceNode));}
   }

   //varyingNotionalInterimExchangePaymentDatesNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* varyingNotionalInterimExchangePaymentDatesNode = xmlNode->FirstChildElement("varyingNotionalInterimExchangePaymentDates");

   if(varyingNotionalInterimExchangePaymentDatesNode){varyingNotionalInterimExchangePaymentDatesIsNull_ = false;}
   else{varyingNotionalInterimExchangePaymentDatesIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- varyingNotionalInterimExchangePaymentDatesNode , address : " << varyingNotionalInterimExchangePaymentDatesNode << std::endl;
   #endif
   if(varyingNotionalInterimExchangePaymentDatesNode)
   {
      if(varyingNotionalInterimExchangePaymentDatesNode->Attribute("href") || varyingNotionalInterimExchangePaymentDatesNode->Attribute("id"))
      {
          if(varyingNotionalInterimExchangePaymentDatesNode->Attribute("id"))
          {
             varyingNotionalInterimExchangePaymentDatesIDRef_ = varyingNotionalInterimExchangePaymentDatesNode->Attribute("id");
             varyingNotionalInterimExchangePaymentDates_ = boost::shared_ptr<RelativeDateOffset>(new RelativeDateOffset(varyingNotionalInterimExchangePaymentDatesNode));
             varyingNotionalInterimExchangePaymentDates_->setID(varyingNotionalInterimExchangePaymentDatesIDRef_);
             IDManager::instance().setID(varyingNotionalInterimExchangePaymentDatesIDRef_,varyingNotionalInterimExchangePaymentDates_);
          }
          else if(varyingNotionalInterimExchangePaymentDatesNode->Attribute("href")) { varyingNotionalInterimExchangePaymentDatesIDRef_ = varyingNotionalInterimExchangePaymentDatesNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { varyingNotionalInterimExchangePaymentDates_ = boost::shared_ptr<RelativeDateOffset>(new RelativeDateOffset(varyingNotionalInterimExchangePaymentDatesNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<NotionalReference> FxLinkedNotionalSchedule::getConstantNotionalScheduleReference()
{
   if(!this->constantNotionalScheduleReferenceIsNull_){
        if(constantNotionalScheduleReferenceIDRef_ != ""){
             return boost::shared_static_cast<NotionalReference>(IDManager::instance().getID(constantNotionalScheduleReferenceIDRef_));
        }else{
             return this->constantNotionalScheduleReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<NotionalReference>();
   }
}
boost::shared_ptr<XsdTypeDecimal> FxLinkedNotionalSchedule::getInitialValue()
{
   if(!this->initialValueIsNull_){
        if(initialValueIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDecimal>(IDManager::instance().getID(initialValueIDRef_));
        }else{
             return this->initialValue_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDecimal>();
   }
}
boost::shared_ptr<Currency> FxLinkedNotionalSchedule::getVaryingNotionalCurrency()
{
   if(!this->varyingNotionalCurrencyIsNull_){
        if(varyingNotionalCurrencyIDRef_ != ""){
             return boost::shared_static_cast<Currency>(IDManager::instance().getID(varyingNotionalCurrencyIDRef_));
        }else{
             return this->varyingNotionalCurrency_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Currency>();
   }
}
boost::shared_ptr<RelativeDateOffset> FxLinkedNotionalSchedule::getVaryingNotionalFixingDates()
{
   if(!this->varyingNotionalFixingDatesIsNull_){
        if(varyingNotionalFixingDatesIDRef_ != ""){
             return boost::shared_static_cast<RelativeDateOffset>(IDManager::instance().getID(varyingNotionalFixingDatesIDRef_));
        }else{
             return this->varyingNotionalFixingDates_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RelativeDateOffset>();
   }
}
boost::shared_ptr<FxSpotRateSource> FxLinkedNotionalSchedule::getFxSpotRateSource()
{
   if(!this->fxSpotRateSourceIsNull_){
        if(fxSpotRateSourceIDRef_ != ""){
             return boost::shared_static_cast<FxSpotRateSource>(IDManager::instance().getID(fxSpotRateSourceIDRef_));
        }else{
             return this->fxSpotRateSource_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<FxSpotRateSource>();
   }
}
boost::shared_ptr<RelativeDateOffset> FxLinkedNotionalSchedule::getVaryingNotionalInterimExchangePaymentDates()
{
   if(!this->varyingNotionalInterimExchangePaymentDatesIsNull_){
        if(varyingNotionalInterimExchangePaymentDatesIDRef_ != ""){
             return boost::shared_static_cast<RelativeDateOffset>(IDManager::instance().getID(varyingNotionalInterimExchangePaymentDatesIDRef_));
        }else{
             return this->varyingNotionalInterimExchangePaymentDates_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RelativeDateOffset>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-ird-5-4/FxLinkedNotionalSchedule.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class FxLinkedNotionalSchedule : ISerialized
    {
        public FxLinkedNotionalSchedule(XmlNode xmlNode)
        {
            XmlNode constantNotionalScheduleReferenceNode = xmlNode.SelectSingleNode("constantNotionalScheduleReference");
            
            if (constantNotionalScheduleReferenceNode != null)
            {
                if (constantNotionalScheduleReferenceNode.Attributes["href"] != null || constantNotionalScheduleReferenceNode.Attributes["id"] != null) 
                {
                    if (constantNotionalScheduleReferenceNode.Attributes["id"] != null) 
                    {
                        constantNotionalScheduleReferenceIDRef_ = constantNotionalScheduleReferenceNode.Attributes["id"].Value;
                        NotionalReference ob = new NotionalReference(constantNotionalScheduleReferenceNode);
                        IDManager.SetID(constantNotionalScheduleReferenceIDRef_, ob);
                    }
                    else if (constantNotionalScheduleReferenceNode.Attributes["href"] != null)
                    {
                        constantNotionalScheduleReferenceIDRef_ = constantNotionalScheduleReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        constantNotionalScheduleReference_ = new NotionalReference(constantNotionalScheduleReferenceNode);
                    }
                }
                else
                {
                    constantNotionalScheduleReference_ = new NotionalReference(constantNotionalScheduleReferenceNode);
                }
            }
            
        
            XmlNode initialValueNode = xmlNode.SelectSingleNode("initialValue");
            
            if (initialValueNode != null)
            {
                if (initialValueNode.Attributes["href"] != null || initialValueNode.Attributes["id"] != null) 
                {
                    if (initialValueNode.Attributes["id"] != null) 
                    {
                        initialValueIDRef_ = initialValueNode.Attributes["id"].Value;
                        XsdTypeDecimal ob = new XsdTypeDecimal(initialValueNode);
                        IDManager.SetID(initialValueIDRef_, ob);
                    }
                    else if (initialValueNode.Attributes["href"] != null)
                    {
                        initialValueIDRef_ = initialValueNode.Attributes["href"].Value;
                    }
                    else
                    {
                        initialValue_ = new XsdTypeDecimal(initialValueNode);
                    }
                }
                else
                {
                    initialValue_ = new XsdTypeDecimal(initialValueNode);
                }
            }
            
        
            XmlNode varyingNotionalCurrencyNode = xmlNode.SelectSingleNode("varyingNotionalCurrency");
            
            if (varyingNotionalCurrencyNode != null)
            {
                if (varyingNotionalCurrencyNode.Attributes["href"] != null || varyingNotionalCurrencyNode.Attributes["id"] != null) 
                {
                    if (varyingNotionalCurrencyNode.Attributes["id"] != null) 
                    {
                        varyingNotionalCurrencyIDRef_ = varyingNotionalCurrencyNode.Attributes["id"].Value;
                        Currency ob = new Currency(varyingNotionalCurrencyNode);
                        IDManager.SetID(varyingNotionalCurrencyIDRef_, ob);
                    }
                    else if (varyingNotionalCurrencyNode.Attributes["href"] != null)
                    {
                        varyingNotionalCurrencyIDRef_ = varyingNotionalCurrencyNode.Attributes["href"].Value;
                    }
                    else
                    {
                        varyingNotionalCurrency_ = new Currency(varyingNotionalCurrencyNode);
                    }
                }
                else
                {
                    varyingNotionalCurrency_ = new Currency(varyingNotionalCurrencyNode);
                }
            }
            
        
            XmlNode varyingNotionalFixingDatesNode = xmlNode.SelectSingleNode("varyingNotionalFixingDates");
            
            if (varyingNotionalFixingDatesNode != null)
            {
                if (varyingNotionalFixingDatesNode.Attributes["href"] != null || varyingNotionalFixingDatesNode.Attributes["id"] != null) 
                {
                    if (varyingNotionalFixingDatesNode.Attributes["id"] != null) 
                    {
                        varyingNotionalFixingDatesIDRef_ = varyingNotionalFixingDatesNode.Attributes["id"].Value;
                        RelativeDateOffset ob = new RelativeDateOffset(varyingNotionalFixingDatesNode);
                        IDManager.SetID(varyingNotionalFixingDatesIDRef_, ob);
                    }
                    else if (varyingNotionalFixingDatesNode.Attributes["href"] != null)
                    {
                        varyingNotionalFixingDatesIDRef_ = varyingNotionalFixingDatesNode.Attributes["href"].Value;
                    }
                    else
                    {
                        varyingNotionalFixingDates_ = new RelativeDateOffset(varyingNotionalFixingDatesNode);
                    }
                }
                else
                {
                    varyingNotionalFixingDates_ = new RelativeDateOffset(varyingNotionalFixingDatesNode);
                }
            }
            
        
            XmlNode fxSpotRateSourceNode = xmlNode.SelectSingleNode("fxSpotRateSource");
            
            if (fxSpotRateSourceNode != null)
            {
                if (fxSpotRateSourceNode.Attributes["href"] != null || fxSpotRateSourceNode.Attributes["id"] != null) 
                {
                    if (fxSpotRateSourceNode.Attributes["id"] != null) 
                    {
                        fxSpotRateSourceIDRef_ = fxSpotRateSourceNode.Attributes["id"].Value;
                        FxSpotRateSource ob = new FxSpotRateSource(fxSpotRateSourceNode);
                        IDManager.SetID(fxSpotRateSourceIDRef_, ob);
                    }
                    else if (fxSpotRateSourceNode.Attributes["href"] != null)
                    {
                        fxSpotRateSourceIDRef_ = fxSpotRateSourceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        fxSpotRateSource_ = new FxSpotRateSource(fxSpotRateSourceNode);
                    }
                }
                else
                {
                    fxSpotRateSource_ = new FxSpotRateSource(fxSpotRateSourceNode);
                }
            }
            
        
            XmlNode varyingNotionalInterimExchangePaymentDatesNode = xmlNode.SelectSingleNode("varyingNotionalInterimExchangePaymentDates");
            
            if (varyingNotionalInterimExchangePaymentDatesNode != null)
            {
                if (varyingNotionalInterimExchangePaymentDatesNode.Attributes["href"] != null || varyingNotionalInterimExchangePaymentDatesNode.Attributes["id"] != null) 
                {
                    if (varyingNotionalInterimExchangePaymentDatesNode.Attributes["id"] != null) 
                    {
                        varyingNotionalInterimExchangePaymentDatesIDRef_ = varyingNotionalInterimExchangePaymentDatesNode.Attributes["id"].Value;
                        RelativeDateOffset ob = new RelativeDateOffset(varyingNotionalInterimExchangePaymentDatesNode);
                        IDManager.SetID(varyingNotionalInterimExchangePaymentDatesIDRef_, ob);
                    }
                    else if (varyingNotionalInterimExchangePaymentDatesNode.Attributes["href"] != null)
                    {
                        varyingNotionalInterimExchangePaymentDatesIDRef_ = varyingNotionalInterimExchangePaymentDatesNode.Attributes["href"].Value;
                    }
                    else
                    {
                        varyingNotionalInterimExchangePaymentDates_ = new RelativeDateOffset(varyingNotionalInterimExchangePaymentDatesNode);
                    }
                }
                else
                {
                    varyingNotionalInterimExchangePaymentDates_ = new RelativeDateOffset(varyingNotionalInterimExchangePaymentDatesNode);
                }
            }
            
        
        }
        
    
        #region ConstantNotionalScheduleReference_
        private NotionalReference constantNotionalScheduleReference_;
        public NotionalReference ConstantNotionalScheduleReference_
        {
            get
            {
                if (this.constantNotionalScheduleReference_ != null)
                {
                    return this.constantNotionalScheduleReference_; 
                }
                else if (this.constantNotionalScheduleReferenceIDRef_ != null)
                {
                    constantNotionalScheduleReference_ = IDManager.getID(constantNotionalScheduleReferenceIDRef_) as NotionalReference;
                    return this.constantNotionalScheduleReference_; 
                }
                else
                {
                      return this.constantNotionalScheduleReference_; 
                }
            }
            set
            {
                if (this.constantNotionalScheduleReference_ != value)
                {
                    this.constantNotionalScheduleReference_ = value;
                }
            }
        }
        #endregion
        
        public string constantNotionalScheduleReferenceIDRef_ { get; set; }
        #region InitialValue_
        private XsdTypeDecimal initialValue_;
        public XsdTypeDecimal InitialValue_
        {
            get
            {
                if (this.initialValue_ != null)
                {
                    return this.initialValue_; 
                }
                else if (this.initialValueIDRef_ != null)
                {
                    initialValue_ = IDManager.getID(initialValueIDRef_) as XsdTypeDecimal;
                    return this.initialValue_; 
                }
                else
                {
                      return this.initialValue_; 
                }
            }
            set
            {
                if (this.initialValue_ != value)
                {
                    this.initialValue_ = value;
                }
            }
        }
        #endregion
        
        public string initialValueIDRef_ { get; set; }
        #region VaryingNotionalCurrency_
        private Currency varyingNotionalCurrency_;
        public Currency VaryingNotionalCurrency_
        {
            get
            {
                if (this.varyingNotionalCurrency_ != null)
                {
                    return this.varyingNotionalCurrency_; 
                }
                else if (this.varyingNotionalCurrencyIDRef_ != null)
                {
                    varyingNotionalCurrency_ = IDManager.getID(varyingNotionalCurrencyIDRef_) as Currency;
                    return this.varyingNotionalCurrency_; 
                }
                else
                {
                      return this.varyingNotionalCurrency_; 
                }
            }
            set
            {
                if (this.varyingNotionalCurrency_ != value)
                {
                    this.varyingNotionalCurrency_ = value;
                }
            }
        }
        #endregion
        
        public string varyingNotionalCurrencyIDRef_ { get; set; }
        #region VaryingNotionalFixingDates_
        private RelativeDateOffset varyingNotionalFixingDates_;
        public RelativeDateOffset VaryingNotionalFixingDates_
        {
            get
            {
                if (this.varyingNotionalFixingDates_ != null)
                {
                    return this.varyingNotionalFixingDates_; 
                }
                else if (this.varyingNotionalFixingDatesIDRef_ != null)
                {
                    varyingNotionalFixingDates_ = IDManager.getID(varyingNotionalFixingDatesIDRef_) as RelativeDateOffset;
                    return this.varyingNotionalFixingDates_; 
                }
                else
                {
                      return this.varyingNotionalFixingDates_; 
                }
            }
            set
            {
                if (this.varyingNotionalFixingDates_ != value)
                {
                    this.varyingNotionalFixingDates_ = value;
                }
            }
        }
        #endregion
        
        public string varyingNotionalFixingDatesIDRef_ { get; set; }
        #region FxSpotRateSource_
        private FxSpotRateSource fxSpotRateSource_;
        public FxSpotRateSource FxSpotRateSource_
        {
            get
            {
                if (this.fxSpotRateSource_ != null)
                {
                    return this.fxSpotRateSource_; 
                }
                else if (this.fxSpotRateSourceIDRef_ != null)
                {
                    fxSpotRateSource_ = IDManager.getID(fxSpotRateSourceIDRef_) as FxSpotRateSource;
                    return this.fxSpotRateSource_; 
                }
                else
                {
                      return this.fxSpotRateSource_; 
                }
            }
            set
            {
                if (this.fxSpotRateSource_ != value)
                {
                    this.fxSpotRateSource_ = value;
                }
            }
        }
        #endregion
        
        public string fxSpotRateSourceIDRef_ { get; set; }
        #region VaryingNotionalInterimExchangePaymentDates_
        private RelativeDateOffset varyingNotionalInterimExchangePaymentDates_;
        public RelativeDateOffset VaryingNotionalInterimExchangePaymentDates_
        {
            get
            {
                if (this.varyingNotionalInterimExchangePaymentDates_ != null)
                {
                    return this.varyingNotionalInterimExchangePaymentDates_; 
                }
                else if (this.varyingNotionalInterimExchangePaymentDatesIDRef_ != null)
                {
                    varyingNotionalInterimExchangePaymentDates_ = IDManager.getID(varyingNotionalInterimExchangePaymentDatesIDRef_) as RelativeDateOffset;
                    return this.varyingNotionalInterimExchangePaymentDates_; 
                }
                else
                {
                      return this.varyingNotionalInterimExchangePaymentDates_; 
                }
            }
            set
            {
                if (this.varyingNotionalInterimExchangePaymentDates_ != value)
                {
                    this.varyingNotionalInterimExchangePaymentDates_ = value;
                }
            }
        }
        #endregion
        
        public string varyingNotionalInterimExchangePaymentDatesIDRef_ { get; set; }
        
    
        
    
    }
    
}

